src/components/Comments/Comment.js

```diff
@@ -265,19 +272,36 @@ const Comment = ({
           </View>
         </ButtonHightlight>
       </View>
-      <TouchableOpacity testID="textPress" onPress={onTextPress} style={styles.flexStartContainer}>
+      <TouchableOpacity
+        testID="textPress"
+        onPress={onTextPress}
+        style={(styles.flexStartContainer, {width: '100%'})}>
         <ButtonHightlight
           onLongPress={handleOnLongPress}
-          style={styles.flexStartContainer}
-          onPress={onTextPress}>
+          onPress={onTextPress}
+          style={(styles.flexStartContainer, {width: '100%'})}>
           <Text style={styles.post}>{getCaptionWithLinkStyle(comment.data?.text)}</Text>
         </ButtonHightlight>
       </TouchableOpacity>
       <View style={styles.constainerFooter}>
-        {isLast && level >= 2 ? (
+        {isLast && level > 2 ? (
           <View testID="level2" style={styles.gap} />
         ) : (
           <>
+            {level !== 2 && (
+              <TouchableOpacity
+                activeOpacity={1}
+                onPress={() => onPress(comment.id, userName)}
+                testID="memoComment">
+                <ButtonHightlight
+                  onLongPress={handleOnLongPress}
+                  style={[styles.btnReply, styles.iconContainer]}
+                  onPress={() => onPress(comment.id, userName)}>
+                  <MemoCommentReply height={normalize(14)} />
+                  {level === 0 && <Text style={styles.btnReplyText}>Reply</Text>}
+                </ButtonHightlight>
+              </TouchableOpacity>
+            )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The structure and functionality of the component have been significantly changed. The `onPress` function now takes two parameters, `comment.id` and `userName`. Ensure that all components using this one are updated accordingly. Also, make sure that the new styles applied do not break the layout on different screen sizes.
```diff
+       <TouchableOpacity
+         testID="textPress"
+         onPress={onTextPress}
+         style={(styles.flexStartContainer, {width: '100%'})}>
+          <ButtonHightlight
+            onLongPress={handleOnLongPress}
+           onPress={onTextPress}
+           style={(styles.flexStartContainer, {width: '100%'})}>
+            <Text style={styles.post}>{getCaptionWithLinkStyle(comment.data?.text)}</Text>
+          </ButtonHightlight>
+        </TouchableOpacity>
+        <View style={styles.constainerFooter}>
+         {isLast && level > 2 ? (
+            <View testID="level2" style={styles.gap} />
+          ) : (
+            <>
+             {level !== 2 && (
+               <TouchableOpacity
+                 activeOpacity={1}
+                 onPress={() => onPress(comment.id, userName)}
+                 testID="memoComment">
+                 <ButtonHightlight
+                   onLongPress={handleOnLongPress}
+                   style={[styles.btnReply, styles.iconContainer]}
+                   onPress={() => onPress(comment.id, userName)}>
+                   <MemoCommentReply height={normalize(14)} />
+                   {level === 0 && <Text style={styles.btnReplyText}>Reply</Text>}
+                 </ButtonHightlight>
+               </TouchableOpacity>
+             )}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->